app/backend/approaches/promptmanager.py

```diff
@@ -0,0 +1,76 @@
+import json
+import pathlib
+from dataclasses import dataclass
+
+import prompty
+from openai.types.chat import ChatCompletionMessageParam
+
+
+@dataclass
+class RenderedPrompt:
+    all_messages: list[ChatCompletionMessageParam]
+    system_content: str
+    few_shot_messages: list[ChatCompletionMessageParam]
+    past_messages: list[ChatCompletionMessageParam]
+    new_user_content: str
+
+
+class PromptManager:
```

Comment:
I made an abstraction for a PromptManager, since some developers weren't as keen on Prompty as others, but I'm not sure if this abstraction would work for other ways of managing prompts anyway, so it might be a premature abstraction? I could remove it and only have PromptyManager for now.

Reply:
If I do keep only PromptyManager, then I could also just construct it in the init of each approach, instead of constructing it in app.py.

Reply:
wrapping prompty is all good IMO. Also helps for tests